test/unit/classification_test.rb

```diff
@@ -439,6 +439,32 @@ def setup
                  global_param_classification.send(:values_hash))
   end
 
+  test 'smart class parameter of hash with merge_overrides and merge_default should return merge all values' do
+    key = FactoryGirl.create(:lookup_key, :as_smart_class_param,
+                             :override => true, :key_type => 'hash', :merge_overrides => true, :merge_default => true,
+                             :default_value => { :default => 'default' }, :path => "organization\nlocation",
+                             :puppetclass => puppetclasses(:one))
+
+    as_admin do
+      LookupValue.create! :lookup_key_id => key.id,
+                          :match => "location=#{taxonomies(:location1)}",
+                          :value => {:example => {:a => 'test'}},
+                          :use_puppet_default => false
+    end
```

Comment:
could you please apply the same in previous test and add default value there so we test, that default value gets overriden unless merge_default is true?